pono.cpp

```diff
@@ -287,9 +293,15 @@ int main(int argc, char ** argv)
         StaticConeOfInfluence coi(fts, { prop }, pono_options.verbosity_);
       }
 
+      if (!fts.only_curr(prop)) {
+        logger.log(1, "Got next state or input variables in property. "
+                   "Generating a monitor state.");
+        prop = add_prop_monitor(fts, prop);
```

Comment:
I'm wondering if we should put this in the `Prover`. I'm thinking for the API so people don't need to know they should call this before passing to a prover.

Reply:
yeah, that is a problem. But, if we do it in the prover, we have the problem to update different ts (orig_ts, ts, conc_ts) at different levels of the prover class inheritence.  I didn't find a easy solution to that. That is why I did it outside of the prover.  It is not perfect solution. Maybe we should revisit it in the future.